src/axom/sidre/tests/sidre_group.cpp

```diff
@@ -1157,6 +1158,140 @@ TEST(sidre_group,save_load_via_hdf5_ids)
+  {
+     pa0[i] = i;
+     pa1[i] = offset + i*factor;
+  }
+
+  // Save the DataStore's root
+  for (int i = 0 ; i < nprotocols ; ++i)
+  {
+    const std::string file_path = file_path_base + protocols[i];
+    root->save(file_path, protocols[i]);
+  }
+
+  // Restore the original DataStore into a child group
+  for (int i = 0; i < nprotocols; ++i)
+  {
+    // Only restore sidre_hdf5 protocol
+    if(protocols[i] != "sidre_hdf5")
+    {
+      continue;
+    }
+
+    DataStore *dscopy = new DataStore();
+    Group * dsroot = dscopy->getRoot();
+    const std::string file_path = file_path_base + protocols[i];
+
+    const std::string group_base("group_");
+    const std::string group_name = group_base + protocols[i];
+    Group* cg = dsroot->createGroup(group_name);
+
+    if (axom::utilities::filesystem::pathExists(file_path))
+    {
+      std::cout << "loading " << file_path << std::endl;
+      cg->load(file_path, protocols[i]);
+
+      EXPECT_TRUE(cg->isEquivalentTo(root, false));
+    }
+    else
+    {
+      std::cout << "Couldn't load " << file_path << " --- file not there." <<std::endl;
+    }
+
+    delete dscopy;
+  }
+
+  delete ds;
+}
+
+//------------------------------------------------------------------------------
+TEST(sidre_group,save_child_restore_as_root)
+{
+  // We'll save a child Group into a file then restore it as the root of
+  // a new DataStore.
+
+  // Create a DataStore; put some groups and views into it
+  const std::string file_path_base("sidre_save_child_restore_as_root_");
+  DataStore* ds = new DataStore();
+  Group* root = ds->getRoot();
+  Group* child1 = root->createGroup("g_a");
+  Group* child2 = child1->createGroup("g_b");
+  child1->createViewScalar<int>("i0", 1);
+  child1->createViewString("s0", "I am a string");
+
+  const int num_elems = 4;
+  int * pa0 =
+    child2->createViewAndAllocate("a0", INT_ID, num_elems)->getArray();
+  double * pa1 =
+    child2->createViewAndAllocate("a1", FLOAT64_ID, num_elems)->getArray();
+
+  const double factor = 2.3;
+  const double offset = -0.23;
+  for (int i = 0; i < num_elems; ++i)
+  {
+     pa0[i] = i;
+     pa1[i] = offset + i*factor;
+  }
+
+  // Save the DataStore's root into a sidre_hdf5 archive, since that is the
+  // protocol that we support for round-trip
+  for (int i = 0; i < nprotocols; ++i)
+  {
+    const std::string file_path = file_path_base + protocols[i];
+    child1->save(file_path, protocols[i]);
+  }
+
+  // Restore the saved child1 into a root group
+  for (int i = 0; i < nprotocols; ++i)
+  {
+    // Only restore sidre_hdf5 protocol
+    if(protocols[i] != "sidre_hdf5")
+    {
+      continue;
+    }
+
+    DataStore * dscopy = new DataStore();
+    const std::string file_path = file_path_base + protocols[i];
+    if (axom::utilities::filesystem::pathExists(file_path))
+    {
+      dscopy->getRoot()->load(file_path, protocols[i]);
+
+      EXPECT_TRUE(dscopy->getRoot()->isEquivalentTo(child1, false));
```

Comment:
It might be helpful to have the symmetric check (here and above):
```cpp
       EXPECT_TRUE(dscopy->getRoot()->isEquivalentTo(child1, false)); 
       EXPECT_TRUE(child1->isEquivalentTo(dscopy->getRoot(), false)); 
```

Reply:
I'll add the symmetric check here and above.

Reply:
done